parsers/EIA.py

```diff
@@ -470,68 +513,50 @@ def fetch_production_mix(
+            additional_production = [
                 datapoint
-                for datapoint in additional_mix
+                for datapoint in additional_production
                 if datapoint["value"] is not None
             ]
-            for point in additional_mix:
+            for point in additional_production:
                 point.update({"value": point["value"] * percentage})
-            mix = _merge_production_mix([mix, additional_mix])
-        if not mix:
-            continue
-
-        for point in mix:
-            negative_threshold = NEGATIVE_PRODUCTION_THRESHOLDS_TYPE.get(
-                type, NEGATIVE_PRODUCTION_THRESHOLDS_TYPE["default"]
-            )
-
-            if (
-                type != "hydro"
-                and point["value"]
-                and 0 > point["value"] >= negative_threshold
-            ):
-                point["value"] = 0
-
-            if type == "hydro" and point["value"] and point["value"] < 0:
-                point.update(
-                    {
-                        "production": {},  # required by merge_production_outputs()
-                        "storage": {type: point.pop("value")},
-                    }
+                production_mix, storage_mix = create_production_storage(
+                    production_mode, point, negative_threshold
                 )
-            else:
-                point.update(
-                    {
-                        "production": {type: point.pop("value")},
-                        "storage": {},  # required by merge_production_outputs()
-                    }
+                additional_breakdown.append(
+                    zoneKey=zone_key,
+                    datetime=point["datetime"],
+                    production=production_mix,
+                    storage=storage_mix,
+                    source="eia.gov",
                 )
+            all_production_breakdowns.append(additional_breakdown)
 
-            # replace small negative values (>-5) with 0s This is necessary for solar
-            point = validate(point, logger=logger, remove_negative=True)
-        mixes.append(mix)
-
-    if not mixes:
+    if len(all_production_breakdowns) == 0:
         logger.warning(f"No production mix data found for {zone_key}")
-        return []
-
+        return ProductionBreakdownList(logger).to_list()
+    all_production_breakdowns = list(
+        filter(lambda x: len(x.events) > 0, all_production_breakdowns)
+    )
     # Some of the returned mixes could be for older timeframes.
     # Fx the latest oil data could be 6 months old.
     # In this case we want to discard the old data as we won't be able to merge it
-    timeframes = [sorted(map(lambda x: x["datetime"], mix)) for mix in mixes]
+    timeframes = [
+        sorted(map(lambda x: x.datetime, breakdowns.events))
+        for breakdowns in all_production_breakdowns
+        if len(breakdowns.events) > 0
+    ]
     latest_timeframe = max(timeframes, key=lambda x: x[-1])
```

Comment:
I ran the test_parser with the US-SE-SEPA zone and it fails here with the following error:
```
  File "C:\Users\vikto\Kod\electricitymap-contrib\parsers\EIA.py", line 549, in fetch_production_mix    latest_timeframe = max(timeframes, key=lambda x: x[-1])
ValueError: max() arg is an empty sequence
```
Note: It also fails on master but in that case it fails because it returns no data.